src/pages/index.js

```diff
@@ -20,100 +24,212 @@ import {
-/*                 Event Handlers                    */
-/*---------------------------------------------------*/
-
 function handleProfileFormSubmit(formValues) {
-  userInfo.setUserInfo(formValues.name, formValues.bio);
-  profilePopup.close();
-  editProfileFormValidator.disableButton();
+  api
+    .patchProfile(formValues.name, formValues.bio)
+    .then(() => {
+      changeSubmitButton(editModalSubmitButton);
+      userInfo.setUserInfo(formValues.name, formValues.bio);
+      // profilePopup.renderLoading(true);
+      profilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      // profilePopup.renderLoading(false);
+      editProfileFormValidator.disableButton();
+      // editModalSubmitButton.textContent = "Save";
+    });
 }
 
 function handleAddCardFormSubmit(formValues) {
   const name = formValues.title;
-  const altName = formValues.title;
   const link = formValues.url;
-  const cardElement = createCard({ name, altName, link });
-  section.addItem(cardElement);
-  cardPopup.close();
+  api
+    .postCards({ name, link })
+    .then((object) => {
+      changeSubmitButton(addModalSubmitButton);
+      const element = createCard(object);
+      section.addItem(element);
+      cardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      // addModalSubmitButton.textContent = "Create";
+    });
 }
 
 function handleImageClick(data) {
   popupImg.open(data);
 }
 
+function handleConfirmModal(data) {
+  trashConfirmPopup.open();
+  trashConfirmPopup.setSubmitFunction(() => {
+    handleDeleteConfirmModal(data.apiData._id);
+    data.removeCard(data.cardElement);
+  });
+}
+
+function handleDeleteConfirmModal(ID) {
+  cards.then((cards) => {
+    cards.forEach((card) => {
+      if (card._id === ID) {
+        api
+          .deleteCard(card._id)
+          .then(() => {
+            changeSubmitButton(trashModalSubmitButton);
+            trashConfirmPopup.close();
+          })
+          .catch((err) => {
+            console.error(err);
+          })
+          .finally(() => {
+            // trashModalSubmitButton.textContent = "Yes";
+          });
+      }
+    });
+  });
+
+  trashConfirmPopup.close();
+}
+
+function handleLikeIconClick(data) {
+  if (data.apiData.isLiked) {
+    api
+      .deleteCardLike(data.apiData._id)
+      .then(() => {
+        data.changeHeartIcon(data);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .putCardLike(data.apiData._id)
+      .then(() => {
+        data.changeHeartIcon(data);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
 /*---------------------------------------------------*/
 /*                 Event Listeners                   */
 /*---------------------------------------------------*/
 
+profilePenIcon.addEventListener("click", () => {
+  avatarModalSubmitButton.textContent = "Save";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.